### PR TITLE
feat(add-remove-facet): add methods to add and remove facets from SearchParameters configuration

### PR DESCRIPTION
**Pull request by Morhaus**
See #327

### CODE REVIEW

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -960,6 +1009,60 @@ SearchParameters.prototype = {
...
+   * @param {string} facet facet name to remove
+   * @return {SearchParameters}
+   */
+  removeFacet: function removeFacet(facet) {
+    if (!this.isConjunctiveFacet(facet)) {
+      return this;
+    }
+
+    const newFacets = this.facets.slice();
+    newFacets.splice(newFacets.indexOf(facet), 1);
```
Comment (bobylito) [inline src/SearchParameters/index.js]: Why not use `filter` like for `removeTagRefinement`:

```js
      tagRefinements: filter(this.tagRefinements, function(t) { return t !== tag; })
```

File: src/SearchParameters/index.js [hunk 2]
```diff
@@ -960,6 +1009,60 @@ SearchParameters.prototype = {
...
+   * @return {SearchParameters}
+   */
+  removeDisjunctiveFacet: function removeDisjunctiveFacet(facet) {
+    if (!this.isDisjunctiveFacet(facet)) {
+      return this;
+    }
+
+    const newFacets = this.disjunctiveFacets.slice();
+    newFacets.splice(newFacets.indexOf(facet), 1);
+    return this.setQueryParameters({
```
Comment (bobylito) [inline src/SearchParameters/index.js]: What do we do with the already set refinements? Maybe we should clean them up?

File: src/SearchParameters/index.js (hunk 1 above)
Comment (Morhaus) [inline src/SearchParameters/index.js]: I'm used to avoiding using filter as a remove function, but filter does make for cleaner code. 👍

File: src/SearchParameters/index.js (hunk 2 above)
Comment (Morhaus) [inline src/SearchParameters/index.js]: Adding a refinement to a facet that is not present in `facets` throws, so maybe we should throw here too?

Comment (vvo) [inline src/SearchParameters/index.js]: I would prefer as a user to not have to clear all refinements before remove a facet but that may be a bad shortcut?

**Comment by bobylito** [comment]
Great PR :) Only problem is that it targets master :) Can you open a new one that targets develop?

**Pull request closed without merging**